dandiapi/api/views/dandiset.py

```diff
@@ -48,7 +48,16 @@ def filter_queryset(self, request, queryset, view):
                 )[:1]
                 queryset = queryset.annotate(name=Subquery(latest_version.values('metadata__name')))
                 return queryset.order_by(ordering)
-
+            elif ordering.endswith('modified'):
+                # name refers to the name of the most recent version, so a subquery is required
+                latest_version = Version.objects.filter(dandiset=OuterRef('pk')).order_by(
+                    '-created'
+                )[:1]
+                # get the `modified` field of the most recent version
+                queryset = queryset.annotate(
+                    modified_version=Subquery(latest_version.values('modified'))
+                )
+                return queryset.order_by(f'{ordering}_version')
```

Comment:
Does Django allow this? If so I think it's cleaner. If not, maybe leave a note about why you can't just use `modified`.
```suggestion
                # Overwrite the `modified` field of the Dandiset with the most recent Version's
                queryset = queryset.annotate(modified=Subquery(latest_version.values('modified')))
                return queryset.order_by(ordering)
```

Reply:
No, Django throws an error since `Dandiset` already has a `modified` field. I added a comment explaining it.